queue_job/models/queue_job.py

```diff
@@ -161,11 +161,22 @@ def _change_job_state(self, state, result=None):
             job_.store()
 
     @api.multi
-    def button_done(self):
-        result = _('Manually set to done by %s') % self.env.user.name
+    def button_done(self, reason):
```

Comment:
Since it cannot be called by a button with a reason, I don't think it makes sense to keep this naming. The reason should not be mandatory, should be a named argument. Also the named argument makes the calls more clear `action_done(reason=_('foo'))`

Rougly:

```python

@api.multi
def action_done(self, reason=None):
    # conditional message, adds "with reason:" only if we have a reason
    self._change_job_state(DONE, result=result)
    return True

@api.multi
def button_done(self):
    _logger.warning('deprecated, replaced by action_done()...
    return self.action_done()

@api.multi
def button_done_ask_reason(self):
    action = self.env.ref(
        'queue_job.action_set_jobs_done'
    ).read()[0]
    return action
```